subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_iso.c

```diff
@@ -0,0 +1,529 @@
+static int prepare_cb(struct lll_prepare_param *p);
+static int prepare_cb_common(struct lll_prepare_param *p);
+static void isr_tx_create(void *param);
+static void isr_tx_normal(void *param);
+static void isr_tx_common(void *param,
+			  radio_isr_cb_t isr_tx,
+			  radio_isr_cb_t isr_done);
+static void isr_done_create(void *param);
+static void isr_done_term(void *param);
+
+int lll_adv_iso_init(void)
+{
+	int err;
+
+	err = init_reset();
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+int lll_adv_iso_reset(void)
+{
+	int err;
+
+	err = init_reset();
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+void lll_adv_iso_create_prepare(void *param)
+{
+	prepare(param);
+	create_prepare_bh(param);
+}
+
+void lll_adv_iso_prepare(void *param)
+{
+	prepare(param);
+	prepare_bh(param);
+}
+
+static int init_reset(void)
+{
+	return 0;
+}
+
+static void prepare(void *param)
+{
+	struct lll_prepare_param *p;
+	struct lll_adv_iso *lll;
+	uint16_t elapsed;
+	int err;
+
+	err = lll_hfclock_on();
+	LL_ASSERT(err >= 0);
+
+	p = param;
+
+	/* Instants elapsed */
+	elapsed = p->lazy + 1;
+
+	lll = p->param;
+
+	/* Save the (latency + 1) for use in event */
+	lll->latency_prepare += elapsed;
+}
+
+static void create_prepare_bh(void *param)
+{
+	int err;
+
+	/* Invoke common pipeline handling of prepare */
+	err = lll_prepare(lll_is_abort_cb, lll_abort_cb, create_prepare_cb, 0,
+			  param);
+	LL_ASSERT(!err || err == -EINPROGRESS);
+}
+
+static void prepare_bh(void *param)
+{
+	int err;
+
+	/* Invoke common pipeline handling of prepare */
+	err = lll_prepare(lll_is_abort_cb, lll_abort_cb, prepare_cb, 0, param);
+	LL_ASSERT(!err || err == -EINPROGRESS);
+}
+
+static int create_prepare_cb(struct lll_prepare_param *p)
+{
+	int err;
+
+	err = prepare_cb_common(p);
+	if (err) {
+		DEBUG_RADIO_START_A(1);
+		return 0;
+	}
+
+	radio_isr_set(isr_tx_create, p->param);
+
+	DEBUG_RADIO_START_A(1);
+	return 0;
+}
+
+static int prepare_cb(struct lll_prepare_param *p)
+{
+	int err;
+
+	err = prepare_cb_common(p);
+	if (err) {
+		DEBUG_RADIO_START_A(1);
+		return 0;
+	}
+
+	radio_isr_set(isr_tx_normal, p->param);
+
+	DEBUG_RADIO_START_A(1);
+	return 0;
+}
+
+static int prepare_cb_common(struct lll_prepare_param *p)
+{
+	struct lll_adv_iso *lll;
+	uint32_t ticks_at_event;
+	uint32_t ticks_at_start;
+	uint16_t event_counter;
+	uint8_t access_addr[4];
+	uint16_t data_chan_id;
+	uint8_t data_chan_use;
+	uint8_t crc_init[3];
+	struct pdu_bis *pdu;
+	struct ull_hdr *ull;
+	uint32_t remainder;
+	uint32_t start_us;
+	uint8_t phy;
+
+	DEBUG_RADIO_START_A(1);
+
+	lll = p->param;
+
+	/* Deduce the latency */
+	lll->latency_event = lll->latency_prepare - 1;
+
+	/* Calculate the current event counter value */
+	event_counter = (lll->payload_count / lll->bn) +
+			(lll->latency_event * lll->bn);
+
+	/* Update BIS packet counter to next value */
+	lll->payload_count += (lll->latency_prepare * lll->bn);
+
+	/* Reset accumulated latencies */
+	lll->latency_prepare = 0;
```

Comment:
We add `U` to integers for the boolean bit fields. Do we need to/should we do that for the other unsigned integer values such as `latency_prepare` as well?